recipes/newrelic/infrastructure/logs/logs.yml

```diff
@@ -126,15 +124,28 @@ install:
             done
           fi
 
+          discoveredLogFiles=$(echo -n "{{.DISCOVERED_LOG_FILES | trim}}")
+          discoveredLogsStringLength=${#discoveredLogFiles}
+
           # If the user provides log files that don't exist and we haven't discovered any log files,
           # we cannot proceed with the installation.
           if [ $foundLogFilesCount -eq 0 ] && { [ $discoveredLogsStringLength -eq 0 ] || [ "$discoveredLogFiles" = "logs: []" ]; }; then
             echo "ERROR Provided log files do not exist: $logFiles. No log file(s) found or discovered. Cannot proceed. Exiting installation." >> /dev/stderr
             exit 16
           fi
 
+          # Build the discovered.yml logs config file used for Docker logs
+          IFS=', ' read -r discoveredLogFilesArray <<< "$discoveredLogFiles"
```

Comment:
Is modifying the IFS here necessary?

Reply:
Why do we need this? We should just need to write the config file?

Reply:
In this case, probably not since it's only one path, but what if we have a string of comma-separated file paths? I guess modifying `IFS` for this can have side effects, so is there a better way to handle this? I'm not exactly a shell expert 😉

Reply:
You might check the block at line 120 that we use for the other log file pattern.  It looks something like this:
```bash
for i in $(echo -n {{.LOG_FILES}} | sed "s/,/ /g"); do
done
```
So if there was a comma, it is now a space, and so the list would look like `/file1 /file2 /file3` which extrapolates to this: `for x in /file1 /file2 /file3; do...` which I think covers the bases.  It isn't very clean but it should work.  In that case, it doesn't become an array, but you are just passing a series of arguments to the for loop, which can be easier to work with (and read) than an array.  Just a thought though.

Reply:
The description in the PR explains the oddity around this with the `sh` library that go-task uses.

Reply:
Yeah, I wasn't thinking about expansion.

Reply:
I'm not sure what this does, and therefore don't see why we need it. My incline would be to remove, particularly knowing that each entry may be a pattern and not an existing file path

Reply:
If we have a csv list of paths (regardless if it contains patterns or full paths), we need to tell bash to use the comma as a separator when reading the string, otherwise the list must be space-delimited which introduces inconsistencies in how set these paths in the recipes. Another issue is that I we can't use the `-d` (delmiter) option with `read` because it's not supported by the `sh` library. Otherwise I could use that and it would read something similar to `read -d "," myArr <<< "$discoveredLogFiles`, but unfortunately it's not possible with the `sh` lib.